File: 9.PalindromeNumber/palindrome.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        xStr = str(x)
        length = len(xStr)
        for i in range(length / 2):
            if xStr[i] != xStr[length - i - 1]:
                return False
        return True


if __name__ == '__main__':
    x = 123321
    print Solution().isPalindrome(1321)

File: 9.PalindromeNumber/palindrome.cc
#include <iostream>
#include <stream>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        stringstream stream;
        string s;
        stream << x;
        stream >> s;
        for (int i = 0; i < s.size() / 2; i++) {
            if (s[i] != s[s.size() - i - 1]) {
                return false;
            }
        }
        return true;
    }
};

int main() {
    int x = 123321;
    Solution s = Solution();
    cout << s.isPalindrome(x) << endl;
}
